crates/ntx-builder/Cargo.toml

```diff
@@ -16,6 +16,7 @@ workspace = true
 [dependencies]
 anyhow                     = { workspace = true }
 futures                    = { workspace = true }
+lru                        = { default-features = false, version = "0.16" }
```

Comment:
nit: maybe we should put this at the workspace level? (I anticipate we'll need LRU cache in other places too).